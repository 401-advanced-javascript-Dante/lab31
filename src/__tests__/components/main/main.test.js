/* eslint-disable no-unused-vars */
import React from 'react';
import {shallow , mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../../../components/form/form.js';


describe('<Form />', () => {

  it('already exist in our app' , () => {
    let test = shallow(<Form/>);
    expect(test.find('button').exists()).toBeTruthy();
  });

  it('Assert state changes properly', () => {
    let test = mount(<Form/>);
    let addButton= test.find('#get');
    addButton.simulate('click');
    expect(test.state('option')).toBe('get');
  });

  it('Assert that state is being transferred to the DOM', () => {
    let test = mount(<Form/>);
    let button= test.find('#post');
    button.simulate('click');
    expect(test.find('p').text()).toBe('Selected REST: post');
  });

  it('Assert DOM stability via snapshot', ()=> {
    let test = renderer.create(<Form />);
    expect(test).toMatchSnapshot();
  });



});