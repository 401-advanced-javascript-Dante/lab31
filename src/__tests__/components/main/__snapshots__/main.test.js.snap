// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Assert DOM stability via snapshot 1`] = `
Array [
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Name :
        <input
          name="name"
        />
      </label>
      <br />
      <label>
        Note :
        <textarea
          name="note"
        />
      </label>
      <br />
      <label>
        complete
        <input
          name="status"
          type="radio"
          value="complete"
        />
      </label>
      <label>
        Incomplete
        <input
          name="status"
          type="radio"
          value="Incomplete"
        />
      </label>
      <br />
      <label>
        difficulty:
        <select
          name="difficulty"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
        </select>
      </label>
      <br />
      <button
        id="sub"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
  <br />,
  <div>
    <h2>
      ToDo List:
    </h2>
    <ul />
  </div>,
]
`;
